ztrouy/frontend-capstone-api
# Feature - Join/Leave Groups

### Purpose
To allow Users to join and leave Groups

### Files Changed
- Updated get_common_games to prevent bug where stale caches were being returned after Members joined or left in `group.py`
- Added views for joining and leaving a Group in `groups.py`
- Added Migrations directory to `.gitignore`

### To Test
Fetch, setup by running `pipenv shell` and `./seed_database.sh` in project root directory, run in the debugger, then confirm the following
- POST endpoint `/groups/{id}/join` adds the User to the appropriate Group if they were not already a member
- DELETE endpoint `/groups/{id}/leave` removes the User from the appropriate Group if they are currently a member
- GET endpoint `/groups/{id}` reflects the changes made by either previous endpoints

closes #18
